### ISSUE TITLE
escape.ident should quote identifiers that have capital letters

### ISSUE THREAD

**Issue by jedwards1211** [segmentio/pg-escape#22]
Expected behavior: `escape.ident('Foo') === '"Foo"'`
Actual behavior: `escape.ident('Foo') === 'Foo'`

If we had ``pool.query(`CREATE DATABASE ${escape.ident(name)}`)``, and `name` is `'Foo'`, the created database would have the name `foo` instead since Postgres lowercases unquoted identifiers.

I'm not even sure if `name.toLowerCase() !== name` would be a sufficient to decide whether to quote because who knows if JS lowercases all the same non-Latin characters as Postgres does?  I think the safest bet would be to add double quotes if the name doesn't match `/^[a-z_][a-z0-9_$]*$/` (case sensitive)

**Comment by sMoKeSz** [comment]
hi. why is this not merged yet ?

**Comment by jedwards1211** [comment]
actually I don't know why I bothered making a PR, the last commit was in 2015 so project is probably dead

**Comment by sMoKeSz** [comment]
something is fishy cause I see allot of weekly downloads 19,017 , that means people is using it.
I made myself a fork so I can use in my project the case sensitive fix
I saw another PR just like yours sitting there from 2015.

https://github.com/segmentio/pg-escape/pull/12

**Comment by jedwards1211** [comment]
yeah, though people can keep using an abandoned package, it happens

**Comment by jedwards1211** [comment]
also that's funny that someone already made a PR, wish I had noticed 😂
